internal/ethapi/api.go

```diff
@@ -1013,6 +1013,14 @@ func (e *revertError) ErrorData() interface{} {
 // Note, this function doesn't make and changes in the state/blockchain and is
 // useful to execute and retrieve values.
 func (s *BlockChainAPI) Call(ctx context.Context, args TransactionArgs, blockNrOrHash rpc.BlockNumberOrHash, overrides *StateOverride) (hexutil.Bytes, error) {
+	num, ok := blockNrOrHash.Number()
+	genesis := s.b.GetGenesis()
+	if ok && uint64(num) < genesis.Number().Uint64() {
+		var result hexutil.Bytes
+		err := s.b.GetSequencerRPCService().CallContext(ctx, &result, "eth_call", args, blockNrOrHash, overrides)
+		return result, err
+	}
+
 	result, err := DoCall(ctx, s.b, args, blockNrOrHash, overrides, s.b.RPCEVMTimeout(), s.b.RPCGasCap())
```

Comment:
We also need to handle historical block-hashes, of which we don't know immediately if they're old or new. Maybe define a generic helper function that takes `ctx, blockNrOrHash` and returns the default function `DoCall`/`DoEstimateGas` or a function that calls the RPC service?

Or alternatively, maybe less verbose: try the local call first, and only if it fails with a not-found error, then do the fallback call. That way we don't have to first check whether or not we know the block. And it's probably easier to abstract across different methods too (call, estimate gas, etc.).

Reply:
I've tried implementing your second suggestion here, for only `EstimateGas`. How does it look?

https://github.com/ethereum-optimism/op-geth/blob/bf7ecd566bd6fb2968a7bf754fe032600be92fcf/internal/ethapi/api.go#L1154-L1161